src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/AddDisableRuntimeMarshallingAttributeFixerTests.cs

```diff
@@ -0,0 +1,165 @@
+
+[assembly: AssemblyMetadata(""MyMetadata"", ""Value"")]
+[assembly: System.Runtime.CompilerServices.DisableRuntimeMarshalling]
+";
+
+            var diagnostic = VerifyCS.Diagnostic(GeneratorDiagnostics.Ids.TypeNotSupported).WithLocation(0).WithArguments("S", "s");
+            await VerifyCodeFixAsync(source, propertiesFile, expectedPropertiesFile, diagnostic);
+        }
+
+        private static async Task VerifyCodeFixAsync(string source, string? propertiesFile, string? expectedPropertiesFile, DiagnosticResult diagnostic)
+        {
+            var test = new Test();
+            test.TestCode = source;
+            test.FixedCode = source;
+            test.BatchFixedCode = source;
+            test.ExpectedDiagnostics.Add(diagnostic);
+            if (propertiesFile is not null)
+            {
+                test.TestState.Sources.Add(($"{Test.FilePathPrefix}Properties{Path.DirectorySeparatorChar}AssemblyInfo.cs", propertiesFile));
+            }
+            if (expectedPropertiesFile is not null)
+            {
+                test.FixedState.Sources.Add(($"{Test.FilePathPrefix}Properties{Path.DirectorySeparatorChar}AssemblyInfo.cs", expectedPropertiesFile));
+                test.BatchFixedState.Sources.Add(($"{Test.FilePathPrefix}Properties{Path.DirectorySeparatorChar}AssemblyInfo.cs", expectedPropertiesFile));
+            }
+            await test.RunAsync();
+        }
+
+        class Test : VerifyCS.Test
+        {
+            public const string FilePathPrefix = "/Project/";
+
+            protected override string DefaultFilePathPrefix => FilePathPrefix;
+        }
+
+        // The Roslyn SDK doesn't provide a good test harness for testing a code fix that triggers
+        // on a source-generator-introduced diagnostic. This analyzer does a decent enough job of triggering
+        // the specific diagnostic in the right place for us to test the code fix.
+        [DiagnosticAnalyzer(LanguageNames.CSharp)]
+        public class MockAnalyzer : DiagnosticAnalyzer
+        {
+            private static readonly DiagnosticDescriptor AddDisableRuntimeMarshallingAttributeRule = GeneratorDiagnostics.ParameterTypeNotSupported;
+
+            public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(AddDisableRuntimeMarshallingAttributeRule);
+
+            public override void Initialize(AnalysisContext context)
+            {
+                context.EnableConcurrentExecution();
+                context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+                context.RegisterSymbolAction(context =>
+                {
+                    var symbol = (IParameterSymbol)context.Symbol;
+
+                    if (context.Symbol.ContainingAssembly.GetAttributes().Any(attr => attr.AttributeClass!.ToDisplayString() == TypeNames.System_Runtime_CompilerServices_DisableRuntimeMarshallingAttribute))
```

Comment:
This check here hard coded because of what we are testing right?

Reply:
Yep. No reason to make this more dynamic since it's just for these tests.